File: Backend/Travle.Model/Requests/DestinationImageRequest.cs
namespace Travle.Model.Requests
{
    /// <summary>
    /// A new image attached to a destination on submit. <see cref="Data"/> is the raw bytes (a base64
    /// string over the wire); the server verifies them against <see cref="ContentType"/> by magic bytes
    /// and generates the thumbnail — the client never supplies a thumbnail (rule 3).
    /// </summary>
    public class DestinationImageRequest
    {
        public byte[]? Data { get; set; }
        public string? ContentType { get; set; }
        public int SortOrder { get; set; }
    }
}

File: UI/travle_core/lib/src/models/destination_image_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'destination_image_request.g.dart';

/// A new image attached to a destination on submit (mirrors the backend
/// `DestinationImageRequest`). [data] is base64 bytes (the server maps it back to
/// `byte[]`, verifies the magic bytes against [contentType], and generates the
/// thumbnail — the client never supplies one).
@JsonSerializable()
class DestinationImageRequest {
  DestinationImageRequest({
    required this.data,
    required this.contentType,
    required this.sortOrder,
  });

  final String data;
  final String contentType;
  final int sortOrder;

  factory DestinationImageRequest.fromJson(Map<String, dynamic> json) =>
      _$DestinationImageRequestFromJson(json);

  Map<String, dynamic> toJson() => _$DestinationImageRequestToJson(this);
}
